apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -1,21 +1,36 @@
 import { Injectable } from '@nestjs/common';
+
+import { SelectIntegration, SelectIntegrationCommand } from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
 import { IntegrationEntity } from '@novu/dal';
-import { GetDecryptedIntegrations, GetDecryptedIntegrationsCommand } from '@novu/application-generic';
 
-import { GetIntegrationsCommand } from '../get-integrations/get-integrations.command';
+import { GetActiveIntegrationsCommand } from './get-active-integration.command';
 
 @Injectable()
 export class GetActiveIntegrations {
-  constructor(private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations) {}
+  constructor(private selectIntegration: SelectIntegration) {}
+
+  async execute(command: GetActiveIntegrationsCommand): Promise<IntegrationEntity[]> {
+    const channelTypes = Object.values(ChannelTypeEnum);
 
-  async execute(command: GetIntegrationsCommand): Promise<IntegrationEntity[]> {
-    return await this.getDecryptedIntegrationsUsecase.execute(
-      GetDecryptedIntegrationsCommand.create({
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-        active: true,
-        userId: command.userId,
+    const integrations = await Promise.all(
+      channelTypes.map(async (channelType) => {
+        return await this.selectIntegration.execute(
+          SelectIntegrationCommand.create({
+            environmentId: command.environmentId,
+            organizationId: command.organizationId,
+            userId: command.userId,
+            channelType: channelType,
+            providerId: command.providerId,
+          })
+        );
       })
     );
+
+    return integrations.filter(notEmpty);
   }
 }
+
+export function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+  return value !== null && value !== undefined;
+}
```

Comment:
Added util to help TS understand that the type is not undefined.
https://github.com/microsoft/TypeScript/issues/45097

Maybe we can move it to shared later.

Reply:
`value is TValue` - do most of the magic there